File: src/main/java/org/springframework/security/web/server/context/ServerWebExchangeAttributeSecurityContextRepositoryFix.java
package org.springframework.security.web.server.context;

import org.springframework.security.core.context.SecurityContext;
import org.springframework.web.server.ServerWebExchange;
import reactor.core.publisher.Mono;

import java.util.Optional;

public class ServerWebExchangeAttributeSecurityContextRepositoryFix implements SecurityContextRepository {

    final String ATTR = "USER";

    public Mono<ServerWebExchange> save(ServerWebExchange exchange, SecurityContext context) {
        exchange.getAttributes().put(ATTR, context);
        return Mono.just(new SecurityContextRepositoryServerWebExchange(exchange, this));
    }

    public Mono<SecurityContext> load(ServerWebExchange exchange) {
        return Mono.justOrEmpty(Optional.ofNullable(exchange.getAttribute(ATTR)));
    }
}

File: src/main/java/org/springframework/security/web/server/context/ServerWebExchangeAttributeSecurityContextRepositoryFix.kt
package org.springframework.security.web.server.context

import org.springframework.security.core.context.SecurityContext
import org.springframework.web.server.ServerWebExchange
import reactor.core.publisher.Mono

import java.util.Optional

class ServerWebExchangeAttributeSecurityContextRepositoryFix : SecurityContextRepository {

    internal val ATTR = "USER"

    override fun save(exchange: ServerWebExchange, context: SecurityContext): Mono<ServerWebExchange> {
        exchange.attributes.put(ATTR, context)
        return Mono.just(SecurityContextRepositoryServerWebExchange(exchange, this))
    }

    override fun load(exchange: ServerWebExchange) = Mono.justOrEmpty(
            Optional.ofNullable(exchange.getAttribute<SecurityContext>(ATTR))
    )

}
